### ISSUE TITLE
`closh.sh` doesn't work on macOS

### ISSUE THREAD

**Issue by mnewt** [dundalek/closh#52]
`closh.sh` uses two features that just about every Unix-like has, but not macOS:
* `realpath`
* `readlink -f`

```
> bin/closh.sh
readlink: illegal option -- f
usage: readlink [-n] [file ...]
usage: dirname path
bin/closh.sh: line 10: realpath: command not found
usage: dirname path
usage: dirname path
bin/closh.sh: line 15: /lumo: No such file or directory
bin/closh.sh: line 15: exec: /lumo: cannot execute: No such file or directory
```

The solution needs to do the same thing with only POSIX tools. A refactor shouldn't be too terrible although it might be harder to cover edge cases. While we're at it, a test or two would probably be good.

I'll send a pull request when I get the chance.

**Comment by domparry** [comment]
Thank you!

**Comment by dundalek** [comment]
Using a shell bootstrapping is an intermediate hack before I am able to compile closh as a single binary. After that it will no longer be needed. So if it is easy, it would be nice so that the latest version runs for people on Mac. Thanks @mnewt

**Comment by mnewt** [comment]
Ok, was going to ask about this. I’ll hold off if you think the nexe packaging will work. It’s not too much work but the more crude posix tools make more potential edge cases which need to be handled.

**Comment by dundalek** [comment]
@mnewt I tried to move forward on the single binary build, but I finally give up. I keep running into a non-ending stream of issues. So I will wait for this until the ecosystem matures a bit or another opportunity presents itself. For example there have been some developments on the clj CLI tool recently which may end up being useful.

In the mean time I will proclaim Windows non-supported with a call for help. If you could port the startup script or find another way how to run on macOS I would be grateful.

**Comment by mnewt** [comment]
Fixed in commit cde9406
